src/AutoRest.CSharp/Common/Output/Models/Types/ObjectType.cs

```diff
@@ -96,7 +96,10 @@ public virtual FormattableString CreateExtraDescriptionWithDiscriminator()
                 List<FormattableString> childrenList = new List<FormattableString>();
                 foreach (var implementation in Discriminator.Implementations)
                 {
-                    childrenList.Add($"{implementation.Type.Implementation.Type:C}");
+                    // when the base type is public and the implementation type is not public, we skip it
+                    if (Type.IsPublic && !implementation.Type.IsPublic)
```

Comment:
If `Type.IsPublic` is false, should it be added to the comment? The current code will do so.

Reply:
we really do not care if that is the case. The base is not public then the derived classes must not be public otherwise the code will not compile. Therefore in that case, everything is not public, it really does not matter what we put as comments on those, and I prefer having less code changed in our generated libraries.